actors/eam/src/lib.rs

```diff
@@ -0,0 +1,289 @@
+        ActorError, EAM_ACTOR_ID, INIT_ACTOR_ADDR,
+    },
+    fvm_ipld_blockstore::Blockstore,
+    fvm_ipld_encoding::{strict_bytes, tuple::*, RawBytes},
+    fvm_shared::{
+        address::{Address, Payload, SECP_PUB_LEN},
+        crypto::hash::SupportedHashes,
+        ActorID, MethodNum, METHOD_CONSTRUCTOR,
+    },
+    num_derive::FromPrimitive,
+    num_traits::FromPrimitive,
+};
+
+#[cfg(feature = "fil-actor")]
+fil_actors_runtime::wasm_trampoline!(EamActor);
+
+/// Maximum allowed EVM bytecode size.
+/// The contract code size limit is 24kB.
+const MAX_CODE_SIZE: usize = 24 << 10;
+
+/// TODO double check this
+const Keccack256_ZERO_INPUT_HASH: [u8; 32] =
+    hex_literal::hex!("c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470");
+
+#[derive(FromPrimitive)]
+#[repr(u64)]
+pub enum Method {
+    Constructor = METHOD_CONSTRUCTOR,
+    Create = 2,
+    Create2 = 3,
+    // CreateAccount = 4,
+}
+
+/// Intermediate type between RLP encoding for CREATE
+struct RlpCreateAddress {
+    address: [u8; 20],
+    nonce: u64,
+}
+
+impl rlp::Encodable for RlpCreateAddress {
+    fn rlp_append(&self, s: &mut rlp::RlpStream) {
+        // TODO check if this is correct... I cant read go code well enough to tell
+        s.encoder().encode_value(&self.address);
+        s.append(&self.nonce);
+    }
+}
+
+#[derive(Serialize_tuple, Deserialize_tuple)]
+pub struct CreateParams {
+    #[serde(with = "strict_bytes")]
+    pub initcode: Vec<u8>,
+    pub nonce: u64,
+}
+
+#[derive(Serialize_tuple, Deserialize_tuple)]
+pub struct Create2Params {
+    #[serde(with = "strict_bytes")]
+    pub initcode: Vec<u8>,
+    /// TODO are we hashing with Little Endian bytes
+    #[serde(with = "strict_bytes")]
+    pub salt: [u8; 32],
+}
+
+#[derive(Serialize_tuple, Deserialize_tuple)]
+pub struct InitAccountParams {
+    #[serde(with = "strict_bytes")]
+    pub pubkey: [u8; SECP_PUB_LEN],
+}
+
+#[derive(Serialize_tuple, Deserialize_tuple)]
+pub struct EamReturn {
+    pub actor_id: ActorID,
+    pub robust_address: Address,
+    #[serde(with = "strict_bytes")]
+    pub eth_address: [u8; 20],
+}
+
+impl EamReturn {
+    fn from_exec4(exec4: Exec4Return, eth_address: [u8; 20]) -> Self {
+        Self {
+            actor_id: exec4.id_address.id().unwrap(),
+            robust_address: exec4.robust_address,
+            eth_address,
+        }
+    }
+}
+
+#[derive(Serialize_tuple, Deserialize_tuple)]
+pub struct EvmConstructorParams {
+    /// The actor's "creator" (specified by the EAM).
+    pub creator: [u8; 20],
+    /// The initcode that will construct the new EVM actor.
+    pub initcode: RawBytes,
+}
+
+// fn eth2f4(addr: &[u8]) -> Result<Address, ActorError> {
+//     Address::new_delegated(EAM_ACTOR_ID, addr)
+//         .map_err(|e| ActorError::illegal_argument(e.to_string()))
+// }
+
+fn assert_code_size(code: &[u8]) -> Result<(), ActorError> {
+    (code.len() == MAX_CODE_SIZE).then(|| ()).ok_or(ActorError::illegal_argument(
```

Comment:
```suggestion
    (code.len() == MAX_CODE_SIZE).then_some(()).ok_or(ActorError::illegal_argument(
```

(but really..., this idiom is probably overkill here)

Reply:
yeah, fair.